File: 10190/main.py
def main():
    while True:
        res = []
        boring = False
        try:
            num_one, num_two = map(int, input().split())
            res.append(num_one)
            while num_one > 1:
                if num_one % num_two == 0:
                    res.append(num_one // num_two)
                else:
                    boring = True
                    print("Boring !")
                    break
                num_one //= num_two
            if boring == False:
                for num in res:
                    print(num, end=" ")
                print()
        except EOFError:
            return


if __name__ == "__main__":
    main()

File: 10190/main.cpp
#include<iostream>
#include<vector>
using namespace std;

int main() {
  vector<int> num;
  bool boring = false;
  int num_one, num_two;
  cin >> num_one >> num_two;
  num.push_back(num_one);
  while (num_one > 1) {
    if (num_one % num_two == 0) {
      num.push_back(num_one / num_two);
      num_one /= num_two;
    }
    else {
      boring = true;
      cout << "Boring!" << endl;
      return 0;
    }
  }
  for (size_t i = 0; i < num.size(); i++) {
    cout << num[i] << " ";
  }
  cout << endl;
  return 0;
}